src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs

```diff
@@ -0,0 +1,140 @@
+        public NuGetRestorer(
+            IUnconfiguredProjectVsServices projectVsServices,
+            IVsSolutionRestoreService solutionRestoreService,
+            IActiveConfiguredProjectSubscriptionService activeConfiguredProjectSubscriptionService,
+            ActiveConfiguredProjectsIgnoringTargetFrameworkProvider activeConfiguredProjectsProvider) 
+            : base(projectVsServices.ThreadingService.JoinableTaskContext)
+        {
+            _projectVsServices = projectVsServices;
+            _solutionRestoreService = solutionRestoreService;
+            _activeConfiguredProjectSubscriptionService = activeConfiguredProjectSubscriptionService;
+            _activeConfiguredProjectsProvider = activeConfiguredProjectsProvider;
+        }
+
+        [ProjectAutoLoad(startAfter: ProjectLoadCheckpoint.ProjectFactoryCompleted)]
+        [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+        internal Task OnProjectFactoryCompletedAsync()
+        {
+            // set up a subscription to listen for target framework changes
+            var target = new ActionBlock<IProjectVersionedValue<IProjectSubscriptionUpdate>>(e => OnProjectChangedAsync(e));
+            _targetFrameworkSubscriptionLink = _activeConfiguredProjectSubscriptionService.ProjectRuleSource.SourceBlock.LinkTo(
+                target: target,
+                ruleNames: _targetFrameworkWatchedRules,
+                initialDataAsNew: false, // only reset on subsequent changes
+                suppressVersionOnlyUpdates: true);
+
+            return Task.CompletedTask;
+        }
+
+        private async Task OnProjectChangedAsync(IProjectVersionedValue<IProjectSubscriptionUpdate> update)
+        {
+            if (IsDisposing || IsDisposed)
+                return;
+
+            await InitializeAsync().ConfigureAwait(false);
+
+            // when TargetFrameworks or TargetFrameworkMoniker changes, reset subscriptions so that
+            // any new configured projects are picked up
+            if (HasTargetFrameworkChanged(update))
+            {
+                await ResetSubscriptions().ConfigureAwait(false);
+            }
+        }
+
+        protected override async Task InitializeCoreAsync(CancellationToken cancellationToken)
+        {            
+            await ResetSubscriptions().ConfigureAwait(false);
+        }
+
+        protected override Task DisposeCoreAsync(bool initialized)
+        {
+            _evaluationSubscriptionLink?.Dispose();
+            _targetFrameworkSubscriptionLink?.Dispose();
+            return Task.CompletedTask;
+        }
+
+        private async Task ResetSubscriptions()
+        {
+            _evaluationSubscriptionLink?.Dispose();
+
+            var currentProjects = await _activeConfiguredProjectsProvider.GetActiveConfiguredProjectsAsync().ConfigureAwait(false);
+
+            if (currentProjects.Any())
+            {
+                var sourceLinkOptions = new StandardRuleDataflowLinkOptions
```

Comment:
>sourceLinkOptions [](start = 20, length = 17)

@mavasani FYI I figured out how to link all the configured project subscriptions and pass the correct rules. The solution was to use this StandardRuleDataflowLinkOptions and pass its value to each created SourceBlockAndLink.